Form/DataField/JsonMenuEditorFieldType.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace EMS\CoreBundle\Form\DataField;
+
+use EMS\CoreBundle\Form\Field\AnalyzerPickerType;
+use EMS\CoreBundle\Form\Field\IconPickerType;
+use Symfony\Component\Form\Extension\Core\Type\HiddenType;
+use Symfony\Component\Form\Extension\Core\Type\IntegerType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class JsonMenuEditorFieldType extends DataFieldType
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getLabel()
+    {
+        return 'JSON menu editor field';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getIcon()
+    {
+        return 'fa fa-sitemap';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getParent()
+    {
+        return HiddenType::class;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildView(FormView $view, FormInterface $form, array $options)
+    {
+        $fieldType = $options['metadata'];
```

Comment:
Fieldtype is only used in line 59. Move this line to line 58? And add a /** @var for IDE help?

Reply:
Ok, done